apps/orders/models.py

```diff
@@ -21,6 +23,18 @@ class Order(models.Model):
     delivery_point = models.ForeignKey(
         "products.DeliveryPoint", on_delete=models.SET_NULL, null=True
     )
+    identifier = models.CharField(max_length=30, unique=True)
+
+    def save(self, *args, **kwargs):
+        if not self.identifier:
+            while True:
+                identifier = " ".join(
+                    "".join(random.choices(string.ascii_uppercase + string.digits, k=5))
+                    for _ in range(4)
+                )
+                if not Order.objects.filter(identifier=identifier).exists():
+                    self.identifier = identifier
+        super().save(*args, **kwargs)
```

Comment:
Can you change this loop for "while not self.identifier", maybe that while true escapes in this version of python when the identifier is set but what if this behaviour changes?